iot_verifier/src/runner.rs

```diff
@@ -451,6 +451,13 @@ impl Runner {
                 return Ok(());
             }
         }
+        // write out metrics for any witness which failed verification
+        // TODO: work our approach that doesnt require the prior cloning of
+        // the selected and unselected witnesses vecs
+        // tried to do this directly from the now discarded poc_proto
+        // but could nae get it to get a way past the lack of COPY
+        fire_invalid_witness_metric(&selected_witnesses);
```

Comment:
Looks like these could be combined with Vec::append() and make a single call to fire the metric?

Reply:
I don't get why `fire_invalid_witness_metric` is called for selected_witnesses. they're a mix of valids and invalids right?

Reply:
bot selected and unselected lists can contain invalids.  the function `fire_invalid_witness_metric` filters out the invalids and only fires the metric for those